dsolve systems simple fixes

cbm755 opened the pull request:
Here are some minor fixes, coming out of #9235.

Should be fairly straightforward to review.

cbm755 committed:
dsolve: drop unused symbol

cbm755 committed:
dsolve: fix documentation error for incorrect meaning of nonlinear

(Linear constant-coeff is only a subset of linear).

cbm755 committed:
dsolve: comment typo fix

cbm755 committed:
dsolve: minor docstring typo

cbm755 committed:
dsolve: use Dummys to prevent symbol clash

Filed #9238 to do properly.

cbm755 committed:
dsolve: minor typo fix

cbm755 committed:
dsolve: fix iterable query

Thanks @meghana1995 for this.

debugger22 commented on sympy/solvers/ode.py:
```diff
@@ -1286,8 +1286,8 @@ def classify_sysode(eq, funcs=None, **kwargs):
-    If the coefficient is constant, then the equation is said to be linear and 'is_linear'
-    returns True otherwise 'is_linear' returns False.
+    Note that "linear" here refers to the operator: terms such as x*diff(x,t) are
+    non-linear, whereas terms like sin(t)*diff(x,t) are still linear operators.
```
Perhaps, two backticks around code will improve readability.

cbm755 replied:
thanks.  I haven't edited other parts of that block as I believe others are doing so elsewhere.

cbm755 committed:
dsolve: doc: improve example code quoting

Improve the quoting of example codes in the linear/nonlinear doc
fix 65231c57e8289bcd19d7c0cc482500d2b67f8bd6.

moorepants commented:
+1
